src/psyclone/tests/parse/module_manager_test.py

```diff
@@ -343,3 +344,133 @@ def test_mod_manager_add_ignore_modules():
+    mod_man.add_files("d1/a_mod.f90")
+
+    #
+    # Test various other functions
+    #
+    mod_man.load_all_source_files()
+    mod_man.create_all_fparser_trees()
+    mod_man.create_all_psyir_nodes()
+    dummy = mod_man.all_file_infos
+    assert dummy is not None
+    mod_man.load_all_module_infos(verbose=True)
+    dummy = mod_man.all_module_infos
+    assert dummy is not None
+
+    # Should raise an error that the first module to be processed
+    # was already processed
+    with pytest.raises(KeyError) as einfo:
+        mod_man.load_all_module_infos(verbose=True)
+
+    assert "Module 'a_mod' already processed" in str(einfo.value)
+
+
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance",
+                         "mod_man_test_setup_directories")
+def test_mod_manager_load_all_module_infos_trigger_error_module_read_twice():
+    '''
+    Make particular check for load_all_module_infos():
+    - Reading in the same module twice is triggering an error.
+    '''
+    mod_man = ModuleManager.get()
+
+    #
+    # Test add_files(...)
+    #
+    mod_man.add_files("d1/a_mod.f90")
+
+    mod_man.load_all_module_infos(verbose=True)
+
+    # Should raise an error that the first module to be processed
+    # was already processed
+    with pytest.raises(KeyError) as einfo:
+        mod_man.load_all_module_infos(verbose=True)
+
+    assert "Module 'a_mod' already processed" in str(einfo.value)
+
+
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance")
+def test_mod_manager_load_all_module_infos_trigger_error_file_read_twice():
+    '''
+    Make particular check for load_all_module_infos():
+    - Reading in the same file twice is triggering an error.
+    '''
+    mod_man = ModuleManager.get()
+
+    with open("t_mod.f90", "w", encoding="utf-8") as f_out:
+        f_out.write("\n")   # Just an empty file
+
+    mod_man.add_files("t_mod.f90")
+    mod_man.load_all_module_infos(verbose=True)
+
+    # Should raise an error that the file was already processed
+    with pytest.raises(KeyError) as einfo:
+        mod_man.load_all_module_infos(verbose=True)
+
+    assert "File 't_mod.f90' already processed" in str(einfo.value)
+
+
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance")
+def test_mod_manager_get_all_rec_used_mod_infos_missing_b(capsys):
+    '''
+    Make particular check for
+    get_all_dependencies_recursively_for_module_name():
+    - Create files for modules 'c'
+    - Dependency of modules is c -> b
+    - This will follow some paths in the control flow
+    - It will not complain about a missing 'b'
+    '''
+    mod_man = ModuleManager.get()
+
+    with open("c_b.f90", "w", encoding="utf-8") as f_out:
+        f_out.write("module c\nuse b\nend")   # Just an empty file
+
+    mod_man.add_files("c_b.f90")
+    mod_man.load_all_module_infos(verbose=True)
+
+    # These are soft errors that don't raise Exceptions
```

Comment:
s/These are soft/The missing dependency is a soft/ if that's right? If not, please elaborate a bit.